### ISSUE TITLE
Text messages on pymumble arrive HTML escaped

### ISSUE THREAD

**Issue by mweinelt** [mumble-voip/mumble#4999]
**Describe the bug**
Somewhere between Mumble and pymumble (botamusique) there is a problem with escaping of HTML entities. https://github.com/azlux/botamusique/issues/274

Writing the following to the channel

```
!playlist https://www.youtube.com/watch?v=rl_ndy61ON4&list=PLBotQZ6u3kciAUtzpqyZETJ_loczWF9kb
```

will be received as

```
!playlist <a href="https://www.youtube.com/watch?v=rl_ndy61ON4&amp;list=PLBotQZ6u3kciAUtzpqyZETJ_loczWF9kb">https://www.youtube.com/watch?v=rl_ndy61ON4&amp;list=PLBotQZ6u3kciAUtzpqyZETJ_loczWF9kb</a>
```

This breaks having multiple GET parameters

So now I wonder who in the line of communications does the escaping, and if that is intentional, and if so, who should unescape them.

**Steps to Reproduce**
1. Send to a channel with botamusique in it
  > !playlist https://www.youtube.com/watch?v=rl_ndy61ON4&list=PLBotQZ6u3kciAUtzpqyZETJ_loczWF9kb`

2. Notice how the URL on the console log has HTML escaping (`&amp;`) in it

**Expected behavior**
Text received should not be HTML escaped.

**Desktop (please complete the following information):**
 - OS: NixOS Unstable
 - Version: https://github.com/mumble-voip/mumble/commit/0a31cd1c63ff42d61470ac006a60f96051c279ec

**Additional context**
pymumbles callback looks just like a thin protobuf layer: https://github.com/azlux/pymumble/blob/754b4f4da6746bb988ad7286713879742686d25c/pymumble_py3/mumble.py#L367-L372
downstream issue at botamusique: https://github.com/azlux/botamusique/issues/274
proposed workaround for botamusique: https://github.com/azlux/botamusique/pull/275

**Comment by ZeroAbility** [comment]
If I recall correctly, `&amp` is in reference to [Google AMP](https://amp.dev/) which is mainly for the mobile space or responsive design. If you use a share link or the desktop web link, does it have the same issue? I'm confused because the source link you are posting and what you have listed as the output are not the same. It does not add the `&amp` portion onto the link when I open it in a browser.

**Comment by mweinelt** [comment]
No, `&amp` is the escape sequence for an ampersand (`&`), a separator between HTTP get parameters (think `http://localhost/?var1=foo&var2=bar&var3=baz`), this has no relation to Google AMP at all.

I just updated the source link to be a codeblock, it was a quote before and that messed with how it was printed.

Basically what happens is that a user posts a message

> My website is at http://localhost/test?foo=bar&baz=bang

What mumble seems to do is make it a HTML link.

```
My website is at <a href="http://localhost/test?foo=bar&baz=bang">http://localhost/test?foo=bar&baz=bang</a>
```

The problem is, that it escapes the doulbe quotes and the ampersand, so it actually looks like this:

```
My website is at <a href=\"http://localhost/test?foo=bar&amp;baz=bang\">http://localhost/test?foo=bar&baz=bang</a>
```

**Comment by ZeroAbility** [comment]
> No, &amp is the escape sequence for an ampersand (&), a separator between HTTP get parameters (think http://localhost/?var1=foo&var2=bar&var3=baz), this has no relation to Google AMP at all.

My mistake. 

> `My website is at <a href=\"http://localhost/test?foo=bar&amp;baz=bang\">http://localhost/test?foo=bar&baz=bang</a>`

It does seem like there is some predictive behavior occuring there, possibly from previous issues with links in the past. I'm not seeing any recent issues at or PR's at first glance that would have produced the behavior. Was it working before?

**Comment by mweinelt** [comment]
I couldn't say. I'm only running botamusique for 10 days or so, and am now catching up on issues I have with it. I can also reproduce this with 1.3.4 and https://github.com/mumble-voip/mumble/commit/0a31cd1c63ff42d61470ac006a60f96051c279ec.

Also I was wrong, it does not escape quotes, only the ampersand is an issue I noticed, so it seems to be strictly HTML entities..

**Comment by Krzmbrzl** [comment]
We use Qt's toHtmlEscaped function in a lot of places. Maybe we are doing that here as well which then results in the observed issue.

In any case though my bet would be that this is indeed a bug on our side :point_up:

**Comment by Krzmbrzl** [comment]
After having looked into this, it turned out as I thought: The HTML-escaped URL was used as the href target for Links sent via Mumble. This will be fixed with #5000
